### PR TITLE
fix(#3606): pass head_branch from webhook context to SimpleGitTool

### PR DESCRIPTION
**Pull request by RC918**
## Summary

Fixes Root Cause #30: In Render webhook worker context, the workspace is in detached HEAD state after `git checkout` of the PR commit. This means `git push` cannot determine the correct branch name from local git state, and `GITHUB_HEAD_REF` environment variable is not available (it's only set in GitHub Actions).

This PR passes `head_branch` from `CiFailureContext` (extracted from webhook payload) through the entire pipeline:
- `AutoFixer.run_auto_fix()` → extracts `head_branch` from `CiFailureContext`
- `AutoFixer._run_project_engineer()` → accepts `target_branch` parameter
- `ProjectEngineerAgent.run_task()` → accepts and passes `target_branch`
- `CodeGenerationWorkflow.execute()` → includes `target_branch` in state
- `SimpleGitTool.commit_and_push()` → uses `target_branch` as priority 1 for push

Priority order for branch determination in `SimpleGitTool`:
1. `target_branch` parameter (from webhook context) ← **NEW**
2. Current git branch name
3. `GITHUB_HEAD_REF` environment variable
4. Fail gracefully (never push to protected 'main')

### Updates since last revision
- Refactored branch determination logic per gemini-code-assist review for better readability
- Changed from sequential `if` statements to cleaner `if/elif/else` structure
- Added explicit `target_branch != 'HEAD'` check to handle edge case where `target_branch` could be the literal string `"HEAD"`
- Rebased on main to include PR #3607 (test fix for `security_validated` in test state)

## Review & Testing Checklist for Human

- [ ] **Verify `CiFailureContext.head_branch` is populated**: Check that the webhook normalizer (`core/flow/schema.py`) correctly extracts and stores `head_branch` from the webhook payload
- [ ] **Verify `CiFailureContext.from_dict()` deserializes `head_branch`**: The `from_dict()` method must include `head_branch` in its reconstruction
- [ ] **Deploy to staging and trigger Probe 0 (#3528)**: After merge, deploy to `morningai-backend-v2-stg-worker` and trigger a CI failure to verify the fix works end-to-end

**Recommended test plan:**
1. Merge and deploy to staging worker
2. Trigger Probe 0 (#3528) CI re-run
3. Check staging logs for `[AutoFixer] Using head_branch from CI failure context: <branch_name>`
4. Verify git push succeeds without "GH013: Repository rule violations" error
5. Confirm AutoFixer successfully creates a fix PR

### Notes
- No unit tests added - this is parameter threading through existing code paths
- The fix cannot be tested locally as it's specific to Render webhook worker context
- Pre-existing lint warnings (E402, C901) are unrelated to this change

**Link to Devin run**: https://app.devin.ai/sessions/570bbc753ba34610b6333a610727b001
**Requested by**: Ryan Chen (ryan2939z@gmail.com) / @RC918

### CODE REVIEW

File: agents/dev_agent/dev_agent_wrapper.py [hunk 1]
```diff
@@ -426,41 +430,59 @@ async def commit_and_push(
...
+                    push_target = branch  # Priority 2: current git branch
+                    branch_source = "git branch"
+
+                if not push_target or push_target == 'HEAD':
                     # Detached HEAD case - try to get branch from environment
                     # GITHUB_HEAD_REF is set by GitHub Actions for PR events
-                    target_branch = os.environ.get('GITHUB_HEAD_REF', '')
-                    if target_branch:
+                    push_target = os.environ.get('GITHUB_HEAD_REF', '')
+                    branch_source = "GITHUB_HEAD_REF"
```
[the remark this answers is not in the document]
Comment (RC918) [reply agents/dev_agent/dev_agent_wrapper.py]: Thanks for the suggestion! I've adopted the refactored `if/elif/else` structure with one modification: added `target_branch != 'HEAD'` check to handle the edge case where `target_branch` could be the literal string `"HEAD"` (which indicates detached HEAD state and should fall through to the env-var fallback).

Commit: c000b717
